packages/lib/shim.ts

```diff
@@ -157,19 +157,25 @@ const shim = {
 	},
 
 	fetchWithRetry: async function(fetchFn: Function, options: any = null) {
+		let requestWithProxy = options?.agent?.proxy || false;
 		if (!options) options = {};
 		if (!options.timeout) options.timeout = 1000 * 120; // ms
-		if (!('maxRetry' in options)) options.maxRetry = shim.fetchMaxRetry_;
+		// For maxRetry the file API sets it to 0, but we want to try at least twice in fetch to accommodate proxy retrying
+		if (!('maxRetry' in options)) options.maxRetry = Math.max(shim.fetchMaxRetry_, 1);
```

Comment:
That shouldn't be changed. We want fetchMaxRetry_ to always be respected

Reply:
I’ll explain why I did this, you have maxRetry 5 set above, but the issue you have here is the fileAPI setting the maxRetry. When a file is not there it’s not there I understand that, but fetching a file and fetching something over the network are separate things, I’m not sure why they share the same config. That’s why I set it here without rewriting the global maxRetry.

so there’s a deeper underlying bug here.